storage/innobase/buf/buf0flu.cc

```diff
@@ -1340,8 +1298,11 @@ static void buf_flush_LRU_list_batch(ulint max, bool evict,
     if (state < buf_page_t::READ_FIX && bpage->lock.u_lock_try(true))
     {
       ut_ad(!bpage->is_io_fixed());
-      bool do_evict= evict;
       switch (bpage->oldest_modification()) {
+      case 2:
+        /* LRU flushing will always evict pages of the temporary tablespace,
+        in buf_page_write_complete(). */
```

Comment:
You are right, we should increment `n.evicted` in both cases. For the pages of the temporary tablespace, we might just increment it here, no matter if we can just discard the freed page or if we must initiate a write.

Yesterday, when debugging the `innodb.innodb_buffer_pool_resize` test in #3107 I observed that we’d spend a lot of time waiting on `buf_pool.mutex` in multiple threads of `buf_page_write_complete()` while evicting pages of the temporary tablespace. In that test scenario, it is inevitable, because the pages were subject to shrinking the buffer pool (`buf_pool.n_blocks_to_withdraw` was being changed), but under normal operation we’d better count these evictions in order to avoid an unnecessarily deep `buf_pool.LRU` scan.

Reply:
Addressed in 50fb3250a60ee68c4273f60bb481b413303bbae8.